### ISSUE TITLE
[$250] keyboard hides unexpectedly when adding a new task on the report screen

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#50442]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.46-3
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @jayeshmangwani 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1728308186934659

## Action Performed:
1. Go to any chat and add a new task. (keyboard should open at this point)
2. Tap on the newly created task to view task details.
3. Check the composer.
## Expected Result:
Keyboard should remain open when viewing the task details.

## Actual Result:
Composer is shown as active, but the keyboard is hidden.

## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/5b355b93-c1b0-4690-b0c5-a4777a2f0ead

https://github.com/user-attachments/assets/9c089ddb-e595-425f-b4be-96d690c5e07d

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021844835682397684717</li>
        <li>Upwork Job ID: 1844835682397684717</li>
        <li>Last Price Increase: 2024-11-08</li>
    </ul>
</details>

**Comment by QichenZhu** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Tapping a task on the report screen while the soft keyboard is open will close the keyboard.

### What is the root cause of that problem?

The soft keyboard is closed during the screen transition by the two pieces of code below.

https://github.com/Expensify/App/blob/1c0d2fa378ca4bac2a56489be9beaded89d3d993/src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx#L663

https://github.com/react-navigation/react-navigation/blob/4caf3cb849a8708a36d0dbf22dac432d2ca780be/packages/stack/src/utils/useKeyboardManager.tsx#L51

<img width="902" alt="Screenshot 2024-10-10 at 1 54 19 AM" src="https://github.com/user-attachments/assets/b709032c-c4c1-4b41-85f3-1d7df916aa7f">

<p></p>

### What changes do you think we should make in order to solve the problem?

We don't need to pass the `autoFocus` attribute to the Android native input since [`useAutoFocusInput`](https://github.com/Expensify/App/blob/1c0d2fa378ca4bac2a56489be9beaded89d3d993/src/components/Composer/index.native.tsx#L53) will take care of it after the screen transaction ends.

```TSX
const isAndroidNative = getPlatform() === CONST.PLATFORM.ANDROID;
```

https://github.com/Expensify/App/blob/1c0d2fa378ca4bac2a56489be9beaded89d3d993/src/components/Composer/index.native.tsx#L127

```TSX
autoFocus={isAndroidNative ? undefined : autoFocus}
```

### What alternative solutions did you explore? (Optional)

We could consider disabling auto focus for the task details screen, just like we did for the money request details screen below.

https://github.com/Expensify/App/blob/1c0d2fa378ca4bac2a56489be9beaded89d3d993/src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx#L302-L307

**Comment by QichenZhu** [comment]
**UPDATE:** My [proposal](https://github.com/Expensify/App/issues/50442#issuecomment-2402353408) doesn’t apply to the latest `main`.

`showSoftInputOnFocus` is set to `false` on purpose to fix https://github.com/Expensify/App/issues/10731, so it’s expected that the soft keyboard doesn’t show.

**Comment by johncschuster** [comment]
Looks like we're still waiting on proposals

**Comment by johncschuster** [comment]
Bumping for Melv

**Comment by johncschuster** [comment]
Bump for Melv

**Comment by johncschuster** [comment]
@allgandalf can you take a look at @QichenZhu's proposal and let me know if it's sufficient? If not, I'll drum up more proposals!

**Comment by mollfpr** [comment]
@johncschuster Do you mean me? 😅

I'll review the latest proposal!

**Comment by johncschuster** [comment]
@allgandalf 

> @johncschuster Do you mean me? 😅

.... 🤦 .... yes. Sorry about that! Did you get a chance to review the latest proposal?

**Comment by mollfpr** [comment]
@QichenZhu to be clear the `autoFocus` from the text input conflict with the blur/dismiss from the library and our component?

Also, why do we still need the `autoFocus` property for iOS even though the `useAutoFocusInput` also focuses on the input?

**Comment by mollfpr** [comment]
I repro the issue a while ago, but after pulling the latest `main` is not reproducing.

https://github.com/user-attachments/assets/a36ff3b0-df4d-4d66-9bce-e52cb2edab03

**Comment by QichenZhu** [comment]
> after pulling the latest main is not reproducing

Still reproducible on my end.

https://github.com/user-attachments/assets/92f8d267-ba21-4395-9372-9a424f755316

> to be clear the autoFocus from the text input conflict with the blur/dismiss from the library and our component?
> 
> Also, why do we still need the autoFocus property for iOS even though the useAutoFocusInput also focuses on the input?

@mollfpr I'll look into it.

**Comment by QichenZhu** [comment]
> to be clear the autoFocus from the text input conflict with the blur/dismiss from the library and our component?

@mollfpr Both factors could contribute to the issue. Addressing either one will make the other take effect.

> Also, why do we still need the autoFocus property for iOS even though the useAutoFocusInput also focuses on the input?

The `useAutoFocusInput` hook was initially introduced for the web and recently applied to the native component to address an Android issue (https://github.com/Expensify/App/issues/45008). I'm not sure why it doesn't work on iOS, but I believe it was never intended to.

**Comment by johncschuster** [comment]
@mollfpr bump on the conversation above!

**Comment by johncschuster** [comment]
Bumped in Slack

**Comment by mollfpr** [comment]
The issue is not reproducible because we disabled auto focus for task reports recently https://github.com/Expensify/App/pull/50544.

**Comment by QichenZhu** [comment]
I don't think the root cause has been fixed. I can still reproduce it with other steps.

https://github.com/user-attachments/assets/eeb1da33-1a82-4e67-9137-b5e7ad48cc10

**Comment by mollfpr** [comment]
@QichenZhu Thanks, I'm able to reproduce the issue with the above step.

It's still not clear to me why the issue only happens on Android, and for the solution even though it's working I feel this is like a workaround.

**Comment by QichenZhu** [comment]
@mollfpr

> It's still not clear to me why the issue only happens on Android

I believe it's clear why it doesn't work on Android, isn't it?

iOS has its own issues (https://github.com/Expensify/App/issues/50442#issuecomment-2441629444, https://github.com/Expensify/App/issues/51482) but it just happens to work in this senario for now. I'd love to explore more if it's a meaningful problem.

> for the solution even though it's working I feel this is like a workaround

Focusing with a delay is a common solution in our repo, used in similar issues like https://github.com/Expensify/App/issues/51010#issuecomment-2419429623 and https://github.com/Expensify/App/issues/51279#issuecomment-2430702882.

**Comment by johncschuster** [comment]
@mollfpr bump on the above comment!

**Comment by johncschuster** [comment]
Bump!
